openstack/networking/v2/subnets/requests.go

```diff
@@ -139,6 +140,8 @@ func (opts CreateOpts) ToSubnetCreateMap() (map[string]interface{}, error) {
 	}
 	if opts.GatewayIP != "" {
 		s["gateway_ip"] = opts.GatewayIP
+	} else if opts.NoGateway {
```

Comment:
Yeah, you're right. I was taking the cheap way out by thinking if a user has both set, they have more problems to worry about 😉